scheduler/src/cook/compute_cluster.clj

```diff
@@ -0,0 +1,105 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns cook.compute-cluster
+  (:require         [clojure.tools.logging :as log]
+                    [datomic.api :as d]))
+
+(defn- create-missing-compute-cluster
+  "Create a missing compute-cluster for one thats not yet in the database."
+  [conn compute-cluster]
+  (log/info "Installing a new compute cluster in datomic for " compute-cluster)
+  @(d/transact
+     conn
+     [(assoc compute-cluster :db/id (d/tempid :db.part/user))]))
+
+(defn- mesos-cluster->compute-cluster-map-for-datomic
+  "Given a mesos cluster dictionary, determine the datomic entity it should correspond to."
+  [{:keys [compute-cluster-name framework-id]}]
+  {:compute-cluster/type :compute-cluster.type/mesos
+   :compute-cluster/cluster-name compute-cluster-name
+   :compute-cluster/mesos-framework-id framework-id})
+
+(defn get-mesos-cluster-entity
```

Comment:
I'd prefer this to return the single entity if it exists, or `nil`.

Reply:
I agree. Thansk!